File: Employee_management_frontend/src/Service/DepartmentService.js
import axios from 'axios'

const Department_BASE_REST_API_URL = 'http://localhost:8080/department';

class DepartmentService{

    getAllDepartments(){
        return axios.get(Department_BASE_REST_API_URL + '/show')
    }

    createDepartment(department){
        return axios.post(Department_BASE_REST_API_URL + '/add', department)
    }

    getDepartmentById(Id){
        return axios.get(Department_BASE_REST_API_URL + '/show/' +Id);
    }

    updateDepartment(department){
        return axios.put(Department_BASE_REST_API_URL + '/update', department);
    }

    deleteDepartment(Id){
        return axios.delete(Department_BASE_REST_API_URL + '/delete/' + Id);
    }
    
}

export default new DepartmentService();

File: Employee_management_backend/src/main/java/com/abhi/services/DepartmentService.java
package com.abhi.services;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.abhi.daos.DepartmentRepository;
import com.abhi.pojos.Department;

@Service
public class DepartmentService {
	@Autowired
	DepartmentRepository dr;
		public void saveOrUpdate(Department dept)
		{
			dr.save(dept);
		}
		public List<Department> getAllDepartments()
		{
			List<Department> dept = new ArrayList<Department>();
			dr.findAll().forEach(dept::add);
			return dept;
		}
		public Department getDepartmentById(int deptId)
		{
			return dr.findById(deptId).get();
		}
		public void update(Department dept, int deptId)
		{
			dr.save(dept);
		}
		public void delete(int deptId)
		{
			dr.deleteById(deptId);
		}

}
